build-tools/pause/pause.c

```diff
@@ -17,19 +17,32 @@ limitations under the License.
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
 #include <unistd.h>
 
 static void sigdown(int signo) {
 	psignal(signo, "shutting down, got signal");
 	exit(0);
 }
 
+static void sigreap(int signo) {
+	while (waitpid(-1, NULL, WNOHANG) > 0);
+}
+
 int main() {
-	if (signal(SIGINT, sigdown) == SIG_ERR)
+	if (getpid() != 1) {
+		fprintf(stderr, "Warning: pause should be the first process in a pod\n");
```

Comment:
Fixed the capitalization on the other messages. I'm inclined to keep it a warning rather than taking down the infra container when presented with an unusual situation. Plus it was helpful to be able to run it outside of a container.